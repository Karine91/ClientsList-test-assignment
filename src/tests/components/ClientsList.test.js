import React from "react";
import { shallow, mount } from "enzyme";
import sinon from "sinon";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import ClientsListConnect, { ClientsList } from "../../components/ClientsList";

const createMockStore = configureMockStore([thunk]);

test("should render ClientsList", () => {
  const wrapper = shallow(<ClientsList clients={[]} dispatch={jest.fn()} />);
  expect(wrapper).toMatchSnapshot();
});

test("calls componentDidMoint", () => {
  const store = createMockStore({ filters: { text: "" }, clients: [] });
  sinon.spy(ClientsListConnect.prototype, "componentDidMount");
  const wrapper = mount(<ClientsListConnect store={store} />);
  expect(ClientsListConnect.prototype.componentDidMount.calledOnce).toEqual(
    true
  );
});
