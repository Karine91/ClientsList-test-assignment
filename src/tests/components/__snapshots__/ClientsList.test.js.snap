// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ClientsList 1`] = `
<List
  className="clients-list"
  selection={true}
/>
`;
